node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -160,20 +166,21 @@ func (r OsDeviceConnectivityHelperScsiGeneric) GetMpathDevice(volumeId string, l
 		targetPath = "/dev/disk/by-path/ip*"
 	}
 
+	logger.Debugf("GetMpathDevice: Start concurrent multipath devices search for volume : [%s]", volumeId)
+	ch := make(chan *WaitForMpathResult)
 	for _, arrayIdentifier := range arrayIdentifiers {
-		dp := strings.Join([]string{targetPath, connectivityType, arrayIdentifier, "lun", lunIdStr}, "-")
-		logger.Infof("GetMpathDevice: Get the mpath devices related to connectivityType=%s initiator=%s and lunID=%d : {%v}", connectivityType, arrayIdentifier, lunId, dp)
-		dps, exists, e := r.Helper.WaitForPathToExist(dp, 5, 1)
-		if e != nil {
-			logger.Errorf("GetMpathDevice: No device found error : %v ", e.Error())
-			errStrings = append(errStrings, e.Error())
-		} else if !exists {
-			e := &MultipleDeviceNotFoundForLunError{volumeId, lunId, []string{arrayIdentifier}}
-			logger.Errorf(e.Error())
-			errStrings = append(errStrings, e.Error())
+		go r.waitForMpath(targetPath, connectivityType, arrayIdentifier, lunId, volumeId, ch)
+	}
+
+	for i := 1; i <= len(arrayIdentifiers); i++ {
```

Comment:
I'm not sure I follow.
`i` is not really used *inside* the loop, even though it doesn't cause an error.
I agree that there are "no new variables" on left side of := when using  `for _, __ := range arrayIdentifiers`, but I still think it's better

Reply:
_,_ is better woulod be better but for go  it is compilation error
for with index is ok for it because i is compared and updated in loop line itself so it treats i is used.